File: desktop/src/bloom/calendar/core/messages.ts
/* eslint-disable camelcase */

export type Event = {
  id: string,
  title: string,
  description: string,
  created_at: Date,
  updated_at: Date,
  start_at: Date,
  end_at: Date,
}

export type Events = {
  events: Event[],
}

export type ListEvents = {
  start_at: Date | null,
  end_at: Date | null,
}

export type CreateEvent = {
  title: string,
  description: string,
  start_at: Date,
  end_at: Date,
}

export type DeleteEvent = {
  id: string,
}

File: core/domain/calendar/messages.go
package calendar

import (
	"time"
)

type ListEventsParams struct {
	StartAt *time.Time `json:"startAt"`
	EndAt   *time.Time `json:"endAt"`
}

type Events struct {
	Events []Event `json:"events"`
}

type CreateEventParams struct {
	Title       string    `json:"title"`
	Description string    `json:"description"`
	StartAt     time.Time `json:"startAt"`
	EndAt       time.Time `json:"endAt"`
}

type DeleteEventParams struct {
	ID string `json:"id"`
}
